rfcs/0167-maintainer-account.md

```diff
@@ -0,0 +1,260 @@
+
+Maintainers are encouraged to self select themselves for maintenance of packages and modules inside nixpkgs.
+This provides flexibility for maintainers as they come and go, and lets them take responsibility for one or more areas in nixpkgs.
+It is quite simple for users to add themselves to the [maintainer list](https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix), and they are expected to do so in a commit prior to adding themselves to a package or module.
+
+The existing [maintainers/README.md](https://github.com/NixOS/nixpkgs/tree/master/maintainers/README.md) in nixpkgs, and the associated maintainer list, spell out a few expectations and requirements for maintainers.
+The requirements in these documents may not be exhaustive and they currently allow for maintainers to opt out of one or more pieces of contact information.
+For example, a maintainer may choose to not provide their user information for the primary nixpkgs hosting platform.
+This opting out for one or more individuals can put extra maintenance overhead on the remaining maintainers; it fragments information which may be important for the overall community; and may even hide information from others working on nixpkgs.
+
+# Detailed design
+
+[design]: #detailed-design
+
+At the time of this writing, the primary nixpkgs hosting platform is GitHub, but this RFC is meant to be platform agnostic.
+If in the future nixpkgs moves to an alternative platform, these requirements and expectations should transfer to the new platform as well.
+
+## Update maintainers/README
+
+Update the section `How to lose maintainer status` in the [maintainers/README.md](https://github.com/NixOS/nixpkgs/tree/master/maintainers/README.md)` file:
+
+```md
+### How to become a maintainer
+
+We encourage people who care about a package to assign themselves as a maintainer.
+Commit access to the nixpkgs repository is not required to become a maintainer.
+
+Maintainers are required to have an account on the nixpkgs hosting platform, which is currently GitHub.
+You are expected to maintain an active account on the nixpkgs hosting platform, and are required to provide this information when adding yourself to `maintainer-list.nix`.
+
+In order to do become a maintainer, add yourself to the [`maintainer-list.nix`](./maintainer-list.nix), and then to the desired package's `meta.maintainers` list, and send a PR with the changes.
+
+### How to lose maintainer status
+
+The following spells out reasons why a maintainer may lose this status. The maintainer is welcome to come back at any time once any issues have been resolved.
+
+#### Inactivity
+
+Maintainers who have become inactive on a given package can be removed.
+This helps us keep an accurate view of the state of maintenance in nixpkgs.
+
+The inactivity measure is currently not strictly enforced.
+We would typically look at it if we notice that the author hasn't reacted to package-related notifications for more than 3 months.
```

Comment:
```suggestion
We would typically look at it if we notice that the author hasn't reacted to all unresolved package-related notifications for more than 3 months.
```
This paragraph made me a little confused: 

If there are several maintainers for a package, do they need to all appear when being pinged?
What if someone just missed one or two pings but still responds to others?

Reply:
I'm open to clarifying this, but the wording you're responding to is already in place: https://github.com/NixOS/nixpkgs/tree/e0cf8780daedd2bc39f5696a9cb5b834fe976009/maintainers#how-to-lose-maintainer-status

I do not expect all maintainers to appear for every package PR or Issue, but I do expect them to be involved frequently. I read this to mean they've not engaged in their responsibility _at all_ during the three months. If they respond to any pings then that would reset any counters. (Even though there are technically no "counters" here and would only be looked at if deemed a problem)

I'd probably suggest at least rewording the duration. Some packages have no activity in 3 months, so perhaps something along the lines of

> for more than 3 months, or 2 package releases, whichever is longer.

Reply:
Okay, changing this would be a non-goal